test_wrapper/test_wrapper.go

```diff
@@ -0,0 +1,174 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"time"
+
+	"cloud.google.com/go/storage"
+	junitFormatter "github.com/jstemmer/go-junit-report/formatter"
+	junitParser "github.com/jstemmer/go-junit-report/parser"
+)
+
+const (
+	metadataURLPrefix   = "http://metadata.google.internal/computeMetadata/v1/instance/attributes/"
+	testTxtResult       = "go-test.txt"
+	testResultObject    = "outs/junit_go-test.xml"
+	testBinaryLocalPath = "image_test"
+	artifactPath        = "/artifact"
+)
+
+func main() {
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		log.Fatalf("failed to create cloud storage client: %v", err)
+	}
+
+	testBinaryURL, err := getMetadataAttribute("_test_binary_url")
+	if err != nil {
+		log.Fatalf("failed to get metadata _test_binary_url: %v", err)
+	}
+
+	u, err := url.Parse(testBinaryURL)
+	if err != nil {
+		log.Fatalf("failed to parse gcs url: %v", err)
+	}
+	bucket, object := u.Host, u.Path
+
+	testRun, err := getMetadataAttribute("_test_run")
+	if err != nil {
+		log.Fatalf("failed to get metadata _test_binary_url: %v", err)
+	}
+
+	var testArguments = []string{"-test.v"}
+	if testRun != "" {
+		testArguments = append(testArguments, "-test.run", testRun)
+	}
+
+	workDir := "/test-" + randString(5)
+	if err = os.Mkdir(workDir+artifactPath, 0755); err != nil {
+		log.Fatalf("failed to create artifact dir: %v", err)
+	}
+	if err = os.Mkdir(workDir, 0755); err != nil {
+		log.Fatalf("failed to create work dir: %v", err)
+	}
+	if err = os.Chdir(workDir); err != nil {
+		log.Fatalf("failed to change work dir: %v", err)
+	}
+	if err = downloadGCSObject(ctx, client, bucket, object, testBinaryLocalPath); err != nil {
+		log.Fatalf("failed to download object: %v", err)
+	}
+
+	if err = executeAndSaveOutput(testTxtResult, testBinaryLocalPath, testArguments); err != nil {
+		log.Fatalf("failed to execute test binary: %v", err)
+	}
+
+	testData, err := convertTxtToJunit(testTxtResult)
```

Comment:
this reads a file you JUST wrote. how about passing this function a string or []byte instead